### PR TITLE
news/blog/2022: MGR TTL Cache blog post

### PR DESCRIPTION
**Pull request by pereman2**
Preview: https://mgrcache.ceph.io/en/news/blog/2022/mgr-ttlcache/

Signed-off-by: Pere Diaz Bou <pdiazbou@redhat.com>

### CODE REVIEW

File: src/en/news/blog/2022/mgr-ttlcache/index.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+tags:
+  - "mgr"
+  - "performance"
+  - "python"
+---
+
+## Introduction
+
+For a time there has been arising problems regarding the scalability of the mgr daemon and the underlying tightly
+coupled (to the daemon) structure of python modules that we know as: dashboard, prometheus, pg_autoscaler, orchestrators and many more.
```
Comment (alfonsomthd) [inline src/en/news/blog/2022/mgr-ttlcache/index.md:15]:
```suggestion
Problems have been arising a while ago related to the scalability of the mgr daemon and the underlying tightly
coupled (to the daemon) structure of python modules like dashboard, prometheus, pg_autoscaler, orchestrators and many more.
```

File: src/en/news/blog/2022/mgr-ttlcache/index.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+Cluster data is requested on demand by the modules every certain time in order to keep the cluster
+status updated in these modules. For example, the dashboard will keep requesting health data updates
+to the daemon periodically to display if the cluster is healthy.
+
+There are instances of data structures where scalability is a problem. Osds for example can increase up to thousands, therefore,
+structures like OSDMap will increase proportionally in size and thus, increasing the latency of modules when they try to reach for
```
Comment (alfonsomthd) [inline src/en/news/blog/2022/mgr-ttlcache/index.md:22]:
```suggestion
structures like OSDMap will increase proportionally in size and thus, increasing the latency of modules when they retrieve
```

File: src/en/news/blog/2022/mgr-ttlcache/index.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+![TTL Cache](images/chart.png)
+
+The blue dashed line represents the improvement in latency to retrieve and osdmap, the black line is the usual behaviour and the blue line,
+the results without caching json objects.
+
+Moreover, the cache is available in master/quincy and soon to be backported to pacific. It can be enabled by running:
+
+`ceph config set mgr mgr ttl_cache_expire_seconds 10`
+
+That will set a 10 seconds _time to live_ to each object cached.
```
Comment (alfonsomthd) [inline src/en/news/blog/2022/mgr-ttlcache/index.md:58]:
```suggestion
That will set a 10 seconds _time to live_ to each cached object.
```

File: src/en/news/blog/2022/mgr-ttlcache/index.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+## Conclusion
+
+With a disabled cache the usual behaviour is still faster, nevertheless, we see a clear improvement of almost 3x on larger clusters with
+the cache enabled and using a simple serialization tool like json is.
```
Comment (alfonsomthd) [inline src/en/news/blog/2022/mgr-ttlcache/index.md:63]:
```suggestion
the cache enabled and using a simple serialization tool like json.
```

File: src/en/news/blog/2022/mgr-ttlcache/index.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+The mgr daemon has been unattended for a while and a lot of improvements are needed to have a more scalable mgr. This cache is an improvement
+that alleviates some of the workload, but there is more work to be done addresing scalability, debugabilitty, subinterpeters...
```
Comment (alfonsomthd) [inline src/en/news/blog/2022/mgr-ttlcache/index.md:66]:
```suggestion
that alleviates some of the workload, but there is more work to be done addressing scalability, debuggabilitty, subinterpreters...
```
Also you can end with: `Stay Tuned!` :smile:

**Review by alfonsomthd — APPROVED** [review]

**Review by Thingee — CHANGES_REQUESTED** [review]
Thanks, @pereman2. I appreciate the submission. I have some grammar and spelling corrections, but otherwise, this LGTM! :+1:

**Comment by pereman2** [comment]
@Thingee Thank you, all checked.

**Pull request merged**